File: models/assetmanagement/AssetUsageLog.js
import mongoose from "mongoose";

const AssetUsageLogSchema = new mongoose.Schema(
{
    hospital_id: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branch_id: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    asset_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "AssetMaster"
    },

    used_by: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User"
    },

    usage_start: Date,
    usage_end: Date,

    notes: String

},
{ timestamps: true }
);

export default mongoose.model("AssetUsageLog", AssetUsageLogSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/AssetUsageLog.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "asset_usage_logs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AssetUsageLog extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "asset_id")
    private AssetMaster asset;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "used_by")
    private User usedBy;

    @Column(name = "usage_start")
    private LocalDateTime usageStart;

    @Column(name = "usage_end")
    private LocalDateTime usageEnd;

    private String notes;
}
